src/assets/javascript/analytics/ga4/init.js

```diff
@@ -0,0 +1,44 @@
+/* global window */
+
+window.DI = window.DI || {}
+window.DI.analyticsGa4 = window.DI.analyticsGa4 || {};
+
+(function (analytics) {
+
+  'use strict'
+
+  const init = function() {
+
+    const consentGiven = window.DI.cookies.hasConsentForAnalytics()
+
+    if (consentGiven) {
+      window.DI.core.load(window.DI.analytics.vars.ga4ContainerId)
+
+      initGa4GlobalTrackers()
+    } else {
+      window.addEventListener('cookie-consent', () => window.DI.analyticsGa4.init())
+    }
+  }
+
+  // Initialise trackers for GA4 events which can be tracked at the global page level, such as page_view events
+  const initGa4GlobalTrackers = function () {
+    const trackers = window.DI.analyticsGa4.trackers
```

Comment:
There's a lot of checks here, but if I've understood correctly trackers should only ever be the PageViewTracker object from `src/assets/javascript/analytics/ga4/pageViewTracker.js`.

Have I missed something? Why do we need this logic?

Reply:
This function is meant to check for a number of trackers, I left it as is because we will probably add other trackers in the future, see GOVUK repo for example https://github.com/alphagov/govuk_publishing_components/tree/main/app/assets/javascripts/govuk_publishing_components/analytics-ga4